File: common/anchor/anchor-android/templates/axp/_blank/src/MyPlugin.java
package @PACKAGE_NAME@;

import com.appspresso.api.*;

/**
 * TODO: change package and class name.
 *
 * Appspresso Plugin Android Module
 *
 *
 */
public class @JAVA_NAME@ implements AxPlugin {

	private AxRuntimeContext runtimeContext;

	@Override
	public void activate(AxRuntimeContext runtimeContext) {
		this.runtimeContext = runtimeContext;

		// TODO: addActivityListener
		// TODO: addWebViewListener
	}

	@Override
	public void deactivate(AxRuntimeContext runtimeContext) {
		// TODO: removeActivityListener
		// TODO: removeWebViewListener
		
		this.runtimeContext = null;
	}

	@Override
	public void execute(AxPluginContext context) {
		String method = context.getMethod();

		if ("echo".equals(method)) {
			String message = context.getParamAsString(0, null);
			context.sendResult(message);
		}
		else {
			context.sendError(AxError.NOT_AVAILABLE_ERR);
		}
	}

}

File: common/anchor/anchor-ios/templates/axp/_blank/MyPlugin.m
//
//  @IOS_CLASS@.m
//
//  Copyright 2011 none. All rights reserved.
//

#import "AxRuntimeContext.h"
#import "AxPluginContext.h"
#import "AxError.h"
#import "@IOS_CLASS@.h"

@implementation @IOS_CLASS@

@synthesize runtimeContext = _runtimeContext;

- (void)activate:(NSObject<AxRuntimeContext>*)runtimeContext {
    _runtimeContext = [runtimeContext retain];
}

- (void)deactivate:(NSObject<AxRuntimeContext>*)runtimeContext {
    [_runtimeContext release];
    _runtimeContext = nil;
}

- (void)execute:(id<AxPluginContext>)context {
    NSString* method = [context getMethod];

    if ([method isEqualToString:@"echo"]) {
        NSString* message = [context getParamAsString:(0)];
        [context sendResult:(message)];
    }
    else {
        [context sendError:(AX_NOT_AVAILABLE_ERR)];
    }
}

@end
